pkg/ssh/ssh.go

```diff
@@ -50,22 +51,30 @@ func SetAuthorizedKeys(cfg *config.CloudConfig, withNet bool) error {
 }
 
 func getKey(key string, withNet bool) (string, error) {
-	if !strings.HasPrefix(key, "github:") {
+	providers := map[string]string{
+		"github": "https://github.com/%s.keys",
+		"gitlab": "https://gitlab.com/%s.keys",
+	}
+
+	url, err := url.Parse(key)
+	if err != nil {
+		return "", err
+	} else if url.Scheme == "" {
 		return key, nil
 	}
 
 	if !withNet {
 		return "", nil
+
+	if providerURL, ok := providers[url.Scheme]; ok {
+		key = fmt.Sprintf(providerURL, url.Opaque)
 	}
 
```

Comment:
if `!ok` we should log a warning about unsupported provider and then `return "", nil`

Reply:
If we do that, we would have to add http and https as providers if we want to support importing keys via http(s) urls.

What I opted to do here was to check if it's one of the supported providers otherwise try and treat it as a http(s) url and if the http.Get call fails with "unsupported protocol scheme" then I know it's something other than a http(s) url.

I did this to avoid having to explicitly check for http(s) in the urls because I couldn't figure out a good way to add http(s) to the providers map. I guess we could use a switch statement instead of a providers map.

If we were to add http(s) to the providers map I would do it like this.
```
providers := map[string]string{
	"github": "https://github.com/%s.keys",
	"gitlab": "https://gitlab.com/%s.keys",
	"http": "%s",
	"https": "%s",
}
```
However in this case I would have to use `url.Opaque`  for gitlab/github and `key` for http(s) in the `Sprintf`.